src/sentry/demo/data_population.py

```diff
@@ -87,55 +105,264 @@ def generate_user():
     return get_user_by_id(id_0_offset)
 
 
-def populate_event_on_project(
-    project: Project, file_path: str, dist_function: Callable[[int], int]
-) -> None:
-    with open(file_path) as f:
-        error_json = json.load(f)
+def safe_send_event(data, basic=False):
+    # TODO: make a batched update version of create_sample_event
+    try:
+        if basic:
+            project = data.pop("project")
+            data["timestamp"] = to_timestamp(data["timestamp"])
+            create_sample_event_basic(data, project.id)
+        else:
+            create_sample_event(
+                **data,
+            )
+    except SnubaError:
+        # if snuba fails, just back off and continue
+        logger.info("safe_send_event.snuba_error")
+        time.sleep(0.2)
+
 
+def clean_event(event_json):
     # clear out these fields if they exist
-    fields_to_delete = ["datetime", "location", "title", "event_id", "project"]
+    fields_to_delete = [
+        "datetime",
+        "timestamp",
+        "start_timestamp",
+        "location",
+        "title",
+        "event_id",
+        "project",
+    ]
     for field in fields_to_delete:
-        if field in error_json:
-            del error_json[field]
+        if field in event_json:
+            del event_json[field]
+
+        # delete in spans as well
+        for span in event_json.get("spans", []):
+            if field in span:
+                del span[field]
+
+    # delete request header since they have data that won't match
+    # the generated data
+    request = event_json.get("request")
+    if request and "headers" in request:
+        del request["headers"]
+
+    return event_json
+
+
+def fix_spans(event_json, old_span_id):
+    """
+    This function will uupdate the span IDs and timstampes for a transaction
+    event
```

Comment:
Could we specify a bit more what "updating" is? (sorry I just got personally confused trying to understand this function)
As I understand it, there's two actions for "fixing"
L168-183: Give spans fresh span_ids & update the parent span ids accordingly
L201-237: Update span offsets and durations based on transaction duration